internal/api/api.go

```diff
@@ -216,6 +219,27 @@ func (api *idler) Info(w http.ResponseWriter, r *http.Request, ps httprouter.Par
 	}
 }
 
+func (api *idler) Reset(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	openShiftAPI, openShiftBearerToken, err := api.getURLAndToken(r)
+	if err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("{\"error\": \"%s\"}", err)))
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	err = api.openShiftClient.Reset(openShiftAPI, openShiftBearerToken, ps.ByName("namespace"))
+	if err != nil {
+		log.Error(err)
```

Comment:
how about moving 
```go
log.Error() 
w.WriteHeader .. 
w.Write(err...)
``` 
that repeats to a function  - `writeError(w, code, err)` ?

Reply:
I did this with writeResponse, but it is creating problems int tests.
The fact that we have multiple function doing this thing creates confusions. I will need to modify functions being used for writeError/writeResponse, but that will affect a lot of other code. I am going avoid this change now, and keep it as it is. Will add this in next PR addresssing all the refactor/cleanup required. How does that sound?